src/main/java/it/gov/pagopa/pu/fileshare/service/FileStorerService.java

```diff
@@ -70,9 +73,39 @@ private Path concatenatePaths(String firstPath, String secondPath) {
 
   private void encryptAndSaveFile(MultipartFile file, Path fileLocation)
     throws IOException {
-    try(InputStream is = file.getInputStream();
-      InputStream cipherIs = AESUtils.encrypt(fileEncryptPassword, is)){
+    try (InputStream is = file.getInputStream();
+         InputStream cipherIs = AESUtils.encrypt(fileEncryptPassword, is)) {
       Files.copy(cipherIs, fileLocation, StandardCopyOption.REPLACE_EXISTING);
     }
   }
+
+  public InputStream decryptFile(String filePath, String fileName) {
+    try {
+      // Build the complete file path
+      File file = Paths.get(filePath, fileName).toFile();
+      if (!file.exists() || !file.isFile()) {
+        log.warn("decryptFile - File [{}] not found", file.getAbsolutePath());
+        return null;
+      }
+
+      InputStream inputStream = new FileInputStream(file);
+
+      log.debug("decryptFile - Decrypting file [{}]", file.getName());
+      inputStream = AESUtils.decrypt(fileEncryptPassword, inputStream);
+
+      return inputStream;
+    } catch (IOException e) {
+      log.error("decryptFile - Error while decrypting file [{}]", filePath, e);
+      throw new IllegalStateException("Error while decrypting the file", e);
+    }
+  }
+
+  public Path buildOrganizationBasePath(Long organizationId) {
+    String sharedPath = foldersPathsConfig.getShared();
+    if (sharedPath == null) {
+      throw new IllegalStateException("Shared folder path is not configured.");
+    }
```

Comment:
this check could be executed on the constructor...
This way the application will not even start if it doesn't exists instead of waiting that this method is called to discover it